background.js

```diff
@@ -46,83 +45,88 @@ chrome.runtime.onMessage.addListener(
-		initStorage(rcxDict.prototype.numList[i*2], "true")
-	}
+			// v0.8.6
+			initStorage("onlyreading", false);
+			// v0.8.8
+			if (items.highlight == "yes")
+				items.highlight = "true";
+			if (items.highlight == "no")
+				items.highlight = "false";
+			if (items.textboxhl == "yes")
+				items.textboxhl = "true";
+			if (items.textboxhl == "no")
+				items.textboxhl = "false";
+			if (items.onlyreading == "yes")
+				items.onlyreading = "true";
+			if (items.onlyreading == "no")
+				items.onlyreading = "false";
+			initStorage("copySeparator", "tab");
+			initStorage("maxClipCopyEntries", "7");
+			initStorage("lineEnding", "n");
+			initStorage("minihelp", "true");
+			initStorage("disablekeys", "false");
+			initStorage("kanjicomponents", "true");
 
-    // v0.8.92
-	initStorage("popupDelay", "150");
-	initStorage("showOnKey", "");
-}
+			for (i = 0; i * 2 < rcxDict.prototype.kanjiInfoLabelList.length; i++) {
+				initStorage(rcxDict.prototype.kanjiInfoLabelList[i * 2], "true")
+			}
 
-/** 
-* Initializes the localStorage for the given key. 
-* If the given key is already initialized, nothing happens. 
-* 
-* @author Teo (GD API Guru)
-* @param key The key for which to initialize 
-* @param initialValue Initial value of localStorage on the given key 
-* @return true if a value is assigned or false if nothing happens 
-*/ 
-function initStorage(key, initialValue) {
-  var currentValue = localStorage[key];
-  if (!currentValue) {
-	localStorage[key] = initialValue;
-	return true;
-  }
-  return false;
-}
+			// v0.8.92
+			initStorage("popupDelay", "150");
+			initStorage("showOnKey", "");
+		}
 
-rcxMain.config = {};
-rcxMain.config.css = localStorage["popupcolor"];
-rcxMain.config.highlight = localStorage["highlight"];
-rcxMain.config.textboxhl = localStorage["textboxhl"];
-rcxMain.config.onlyreading = localStorage["onlyreading"];
-rcxMain.config.copySeparator = localStorage["copySeparator"];
-rcxMain.config.maxClipCopyEntries = localStorage["maxClipCopyEntries"];
-rcxMain.config.lineEnding = localStorage["lineEnding"];
-rcxMain.config.minihelp = localStorage["minihelp"];
-rcxMain.config.popupDelay = parseInt(localStorage["popupDelay"]);
-rcxMain.config.disablekeys = localStorage["disablekeys"];
-rcxMain.config.showOnKey = localStorage["showOnKey"];
-rcxMain.config.kanjicomponents = localStorage["kanjicomponents"];
-rcxMain.config.kanjiinfo = new Array(rcxDict.prototype.numList.length/2);
-for (i = 0; i*2 < rcxDict.prototype.numList.length; i++) {
-	rcxMain.config.kanjiinfo[i] = localStorage[rcxDict.prototype.numList[i*2]];
-}
+		/**
+		 * Initializes the localStorage for the given key.
+		 * If the given key is already initialized, nothing happens.
+		 *
+		 * @author Teo (GD API Guru)
+		 * @param key The key for which to initialize
+		 * @param initialValue Initial value of localStorage on the given key
+		 * @return boolean if a value is assigned or false if nothing happens
+		 */
+		function initStorage(key, initialValue) {

```

Comment:
This function should probably be changed to work with cloud sync. (Right now, you read from cloud sync but you don't handle updating it with default values; if we change this, we can have it work with legacy localstorage values and default values)

It can't work exactly the same since you can't write to cloud storage one field at a time, so we'll need to accumulate the set of initial values and then save them back to cloud storage (either always, or only if the final values are different than what we read from the cloud).

The way I would do that would be to reuse `rcxMain.config` since it's the local store for settings anyway. You can initialize it earlier and then update it one field at a time inside of `initStorage`. Then, after all the settings are initialized we can save the settings back to cloud storage as mentioned before.

The high level goal would be to use cloud setting if it exist, then check localStorage, and then just use default value.